react/src/components/AssetDetail/AssetDetail.tsx

```diff
@@ -0,0 +1,172 @@
+import React, { Suspense, useState, useEffect } from 'react';
+import _ from 'lodash';
+import { useAppConfiguration, useFeatureSelection } from '@hazmapper/hooks';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { Asset, FeatureTypeNullable } from '@hazmapper/types';
+import { FeatureIcon } from '@hazmapper/components/FeatureIcon';
+import { Button, LoadingSpinner, SectionMessage } from '@tacc/core-components';
+import styles from './AssetDetail.module.css';
+import { faCameraRetro, faPhotoFilm } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+type AssetModalProps = {
+  isOpen: boolean;
+  projectId: number;
+  isPublicView: boolean;
+};
+
+const AssetDetail: React.FC<AssetModalProps> = ({
+  isOpen,
+  projectId,
+  isPublicView,
+}) => {
+  const { selectedFeature, selectedFeatureId, setSelectedFeatureId } =
+    useFeatureSelection();
+
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  const closePanel = () => {
+    const params = new URLSearchParams(location.search);
+    params.delete('selectedFeature');
+    navigate(`${location.pathname}?${params.toString()}`);
+  };
+
+  const config = useAppConfiguration();
+  const geoapiUrl = config.geoapiUrl;
+
+  const [selectedFeatureAsset, setSelectedFeatureAsset] = useState<
+    Asset | undefined
+  >(selectedFeature?.assets[0]);
+
+  useEffect(() => {
+    const featureAsset = selectedFeature?.assets[0];
+    setSelectedFeatureAsset(featureAsset);
+  }, [selectedFeature]);
+
+  const featureSource: string | undefined =
+    geoapiUrl + '/assets/' + selectedFeatureAsset?.path;
+
+  const fileType: string | undefined = selectedFeatureAsset?.asset_type;
```

Comment:
Can we use getFeatureType from react/src/types/feature.ts here? Since fileType is currently a string but then cast to FeatureTypeNullable, it would be simpler and more consistent to directly use FeatureType. I also noticed some ? icons appearing for features without assets, and this approach could help address that.

Reply:
then selectedFeatureAsset coudl be dropped as relying on getFeatureType for that logic.

Reply:
This was resolved with your helpful suggestions. Thank you!